src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/MlInformation.cs

```diff
@@ -0,0 +1,119 @@
+    {
+        private IntPtr _handle = IntPtr.Zero;
+        private InfoType _type;
+
+        public enum InfoType
+        {
+            Option = 0,
+            Information = 1,
+        }
+
+        MlInformation() {
+            NNStreamer.CheckNNStreamerSupport();
+
+            NNStreamerError ret = Interop.Util.CreateOption(out _handle);
+            NNStreamer.CheckException(ret, "Failed to create information handle");
+
+            _type = InfoType.Option;
+        }
+
+        internal MlInformation(IntPtr handle) {
+            NNStreamer.CheckNNStreamerSupport();
+            if (handle == IntPtr.Zero)
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The information handle is null");
+
+            _handle = handle;
+            _type = InfoType.Information;
+        }
+
+        ~MlInformation() {
+            NNStreamerError ret = NNStreamerError.None;
+            if (_handle != IntPtr.Zero)
+            {
+                switch(_type)
+                {
+                    case InfoType.Option:
+                        ret = Interop.Util.DestroyOption(_handle);
+                        break;
+                    case InfoType.Information:
+                        ret = Interop.Util.DestroyInformation(_handle);
+                        break;
+                }
+            }
+
+            NNStreamer.CheckException(ret, "Failed to destroy the information");
+        }
+
+        public InfoType GetType()
+        {
+            return _type;
+        }
+
+        public void SetInformation(string key, string value)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property key is invalid");
+
+            if (string.IsNullOrEmpty(value))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property value is invalid");
+
+            NNStreamerError ret = NNStreamerError.None;
+            switch(_type)
+            {
+                case InfoType.Option:
+                    IntPtr valuePtr = Interop.Util.StringToIntPtr(value);
+                    ret = Interop.Util.SetOptionValue(_handle, key, valuePtr, null);
+                    break;
+                case InfoType.Information:
+                    ret = NNStreamerError.NotSupported;
+                    Log.Error(NNStreamer.TAG, "InfoType Iniformation does not support set value");
+                    break;
+            }
+
+            NNStreamer.CheckException(ret, "Failed to set option value");
+        }
+
+        public string GetInformation(string key)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property key is invalid");
+
+            IntPtr value = IntPtr.Zero;
+            NNStreamerError ret = NNStreamerError.None;
```

Comment:
ditto, init value to invalid-param or internal error for the case if info-type is invalid.